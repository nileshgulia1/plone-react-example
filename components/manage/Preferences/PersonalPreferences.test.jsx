import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-intl-redux';
import configureStore from 'redux-mock-store';

import PersonalPreferences from './PersonalPreferences';

const mockStore = configureStore();

jest.mock('react-portal', () => ({
  Portal: jest.fn(() => <div id="Portal" />),
}));

describe('PersonalPreferences', () => {
  it('renders a personal preferences component', () => {
    const store = mockStore({
      intl: {
        locale: 'en',
        messages: {},
      },
    });
    const component = renderer.create(
      <Provider store={store}>
        <PersonalPreferences location={{ pathname: '/blog' }} />
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
