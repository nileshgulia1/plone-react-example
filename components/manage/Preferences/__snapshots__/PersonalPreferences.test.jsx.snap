// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalPreferences renders a personal preferences component 1`] = `
<div
  className="ui container"
  id="page-personal-preferences"
>
  <div
    className="ui stackable top attached tabular menu"
  >
    <a
      className="item"
      onClick={[Function]}
      style={Object {}}
    >
      <span>
        Personal Information
      </span>
    </a>
    <div
      className="active item"
      onClick={[Function]}
    >
      Personal Preferences
    </div>
    <a
      className="item"
      onClick={[Function]}
      style={Object {}}
    >
      <span>
        Password
      </span>
    </a>
  </div>
  <div
    className="ui container"
  >
    <form
      action={undefined}
      className="ui form"
      method="post"
      onSubmit={[Function]}
    >
      <div
        className="ui raised segments"
      >
        <div
          className="ui segment"
        >
          <div
            className="inline field help"
          >
            <div
              className="ui grid"
            >
              <div
                className="stretched row"
              >
                <div
                  className="four wide column"
                >
                  <div
                    className="wrapper"
                  >
                    <label
                      htmlFor="field-language"
                    >
                      Language
                    </label>
                  </div>
                </div>
                <div
                  className="eight wide column"
                >
                  <div
                    aria-busy={undefined}
                    aria-disabled={false}
                    aria-expanded={false}
                    aria-multiselectable={undefined}
                    className="ui selection dropdown"
                    id="field-language"
                    name="language"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onMouseDown={[Function]}
                    role="listbox"
                    tabIndex={0}
                  >
                    <div
                      aria-live="polite"
                      className="text"
                      role="alert"
                    >
                      No value
                    </div>
                    <i
                      aria-hidden="true"
                      className="dropdown icon"
                      onClick={[Function]}
                    />
                    <div
                      className="menu transition"
                    >
                      <div
                        aria-checked={true}
                        aria-disabled={undefined}
                        aria-selected={true}
                        className="active selected item"
                        onClick={[Function]}
                        role="option"
                        style={
                          Object {
                            "pointerEvents": "all",
                          }
                        }
                      >
                        <span
                          className="text"
                        >
                          No value
                        </span>
                      </div>
                      <div
                        aria-checked={false}
                        aria-disabled={undefined}
                        aria-selected={false}
                        className="item"
                        onClick={[Function]}
                        role="option"
                        style={
                          Object {
                            "pointerEvents": "all",
                          }
                        }
                      >
                        <span
                          className="text"
                        >
                          English
                        </span>
                      </div>
                      <div
                        aria-checked={false}
                        aria-disabled={undefined}
                        aria-selected={false}
                        className="item"
                        onClick={[Function]}
                        role="option"
                        style={
                          Object {
                            "pointerEvents": "all",
                          }
                        }
                      >
                        <span
                          className="text"
                        >
                          Deutsch
                        </span>
                      </div>
                      <div
                        aria-checked={false}
                        aria-disabled={undefined}
                        aria-selected={false}
                        className="item"
                        onClick={[Function]}
                        role="option"
                        style={
                          Object {
                            "pointerEvents": "all",
                          }
                        }
                      >
                        <span
                          className="text"
                        >
                          Nederlands
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="stretched row"
              >
                <div
                  className="stretched twelve wide column"
                >
                  <p
                    className="help"
                  >
                    Your preferred language
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ui clearing segment actions"
        >
          <button
            className="ui big basic circular icon primary right floated button"
            disabled={undefined}
            onClick={[Function]}
            role="button"
            tabIndex={undefined}
            title="Save"
            type="submit"
          >
            <i
              aria-hidden="true"
              className="arrow right icon"
            />
          </button>
          <button
            className="ui big basic circular icon secondary right floated button"
            disabled={undefined}
            onClick={[Function]}
            role="button"
            tabIndex={undefined}
            title="Cancel"
          >
            <i
              aria-hidden="true"
              className="remove icon"
            />
          </button>
        </div>
      </div>
    </form>
  </div>
   
  <div
    id="Portal"
  />
</div>
`;
